### PR TITLE
Fix issue where realms are hanging in staging

### PR DESCRIPTION
**Pull request by habdelra**
The fix here is that the PK for the boxel_index should include the `realm_url` column, we didn't see this locally because our tests generally use a base realm that lives in a different DB than the realm under test. The unique constraint error caused indexing to break, which meant that the realm never became ready

### CODE REVIEW

**Comment by habdelra** [comment]
I deployed this branch into staging and the realms stopped hanging

File: packages/realm-server/tests/helpers/index.ts [hunk 1]
```diff
@@ -142,30 +144,33 @@ export async function createRealm({
...
+          realmAdapter: adapter,
+          runnerOptsManager: manager,
+          loader: realm.loaderTemplate,
+          indexRunner,
+        });
+        await worker.run();
+      },
     },
-  });
+    { deferStartUp },
```
Comment (habdelra) [inline packages/realm-server/tests/helpers/index.ts:172]: indenting made this diff hard to read, this line was the only change here

Comment (backspace) [reply packages/realm-server/tests/helpers/index.ts:172]: boxel 2024-05-15 20-57-06

boxel 2024-05-15 20-57-59

“Hide whitespace” to the rescue!

File: packages/realm-server/tests/realm-server-test.ts [hunk 2]
```diff
@@ -2183,6 +2185,97 @@ module('Realm Server serving from root', function (hooks) {
...
+    await base.reindex();
+    await testRealm.reindex();
+
+    {
+      let response = await request
+        .get('/demo/person-1')
+        .set('Accept', 'application/vnd.card+json');
+      assert.strictEqual(response.status, 200, 'HTTP 200 status');
+    }
+  });
```
Comment (backspace) [inline packages/realm-server/tests/realm-server-test.ts:2276]: nice new test 🎉

**Review by backspace — APPROVED** [review]

**Review by jurgenwerk — APPROVED** [review]
Nice, thanks for fixing!

**Review by tintinthong — APPROVED** [review]

**Review by FadhlanR — APPROVED** [review]

**Pull request merged**